### PR TITLE
Remove dependency on `downcast_rs`

### PR DESCRIPTION
**Pull request by solson**
This is the result of a short discussion on Discord. We don't seem to use `downcast_rs` in any essential way, and we could just use the standard library `Any` directly.

cc @bartlomieju

### CODE REVIEW

File: core/resources.rs [hunk 1]
```diff
@@ -85,15 +78,6 @@ impl ResourceTable {
   }
 }
 
-/// Abstract type representing resource in Deno.
-///
-/// The only thing it does is implementing `Downcast` trait
-/// that allows to cast resource to concrete type in `TableResource::get`
-/// and `TableResource::get_mut` methods.
-pub trait Resource: Downcast + Any {}
```
Comment (bartlomieju) [inline core/resources.rs:93]: Let's keep `Resource` trait

**Review by bartlomieju — COMMENTED** [review]
@solson great that we can remove dependency on `downcast_rs`!

File: core/resources.rs (hunk 1 above)
Comment (solson) [reply core/resources.rs:93]: In what form? If we do `trait Resource: Any {}`, we run into the problem `Downcast` exists to solve (going from `Box<dyn Resource>` to `Box<dyn Any>`). But this doesn't really gain us anything. What is the desired semantics for `Resource`?

it seems like maybe we just want a "trait alias", but that's a concept that doesn't exist in Rust yet (see https://github.com/rust-lang/rust/issues/41517).

Comment (solson) [reply core/resources.rs:93]: As it currently stands, the resource table really can be used to store "any" Rust value (as long as the user manages their keys and knows which types to `.get::<T>()` with — the usual `Any` caveats).

is there a desire to replace this with something more structured or restricted?

Comment (akritrime) [reply core/resources.rs:93]: I way I see it, its about mapping the mental model of how Deno functions to abstractions in code. Removing the `Resource` trait removes one core piece of the puzzle, where you have a `ResourceTable` and a `ResourceId` but no explicit `Resource`. While in practical uses, `Resource` is the same thing as `Any`, having a `Resource` trait makes reading and understanding the code a bit easier.

Comment (solson) [reply core/resources.rs:93]: If anything, I think we could replace the `(String, Box<dyn Any>)` anonymous tuple in the `ResourceTable` with a named type:

```rust
struct Resource {
    name: String,
    resource: Box<dyn Any>,
}
```

I think the current `Resource` trait is a misleading puzzle piece to me as a Rust programmer — it's an extra layer obscuring the fact that it's really just `Any` with no additional restrictions.

Comment (akritrime) [reply core/resources.rs:93]: I agree with the `Resource` struct bit, which, maybe due to my limited exposure, seems a better way of representing what a `Resource` is.

Comment (ry) [reply core/resources.rs:93]: I like @solson's argument - `Any` is more general and easier to think about. The fewer restrictions, the better.

**Review by ry — APPROVED** [review]
LGTM - nice patch!

**Pull request merged**